examples/dataservice-read/CMakeLists.txt

```diff
@@ -16,16 +16,39 @@
 # License-Filename: LICENSE
 cmake_minimum_required(VERSION 3.5)
 
-if (IOS OR ANDROID)
-    message(WARNING "Unsupported architecture!")
-    return()
-endif()
+set(EDGE_SDK_DATASERVICE_READ_EXAMPLE_TARGET dataservice-read-example)
+
+set(EDGE_SDK_EXAMPLE_SUCCESS_STRING "Reading partition's data from catalog performed successfully")
+set(EDGE_SDK_EXAMPLE_FAILURE_STRING "Failed to read data from the catalog!")
+
+if (IOS)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/../helpers/cmake/gen_ios_example.cmake.in)
+
+
+    gen_ios_example_application(${EDGE_SDK_DATASERVICE_READ_EXAMPLE_TARGET}
+                                ${EDGE_SDK_EXAMPLE_SUCCESS_STRING}
+                                ${EDGE_SDK_EXAMPLE_FAILURE_STRING})
 
-project(dataservice-read-example)
+elseif (ANDROID)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/../helpers/cmake/gen_android_example.cmake.in)
 
-add_executable(dataservice-read-example example.cpp)
+    set(EDGE_SDK_EXAMPLE_PACKAGE_NAME "dataservice_read_example")
+    set(EDGE_SDK_EXAMPLE_PACKAGE_JNI_NAME "dataservice_1read_1example")
```

Comment:
is it done intentionally?

Reply:
You meant about 1s? Yes, it is for JNI - basically, it is how JNI 'translate' JAVA native methods.

Alternatively, we can invoke JNI compiler to generate those methods' names automatically during CMakeLists.txt execution and replace generated names via configure_file(). It is more complex way, but, I guess, it is how it should be done.